packages/core/collections/notebooks.js

```diff
@@ -155,4 +155,28 @@ export default class Notebooks extends Collection {
       await this._db.trash.add(notebookData);
     }
   }
+
+  async moveTopics(id, items) {
+    if (!id) throw new Error("The destination notebook cannot be undefined.");
+    if (!items.length) throw new Error("You must select one or more topics");
+
+    let topicIds = [];
+    let lastNotebookId = items[0].notebookId;
+    let noteIds = [];
+
+    for (let index = 0; index < items.length; index++) {
+      topicIds.push(items[index].id);
+      items[index].notebookId = id;
+      await this._db.notes.addToNotebook(
+        { id, topic: items[index].id },
+        ...this._db.notes.topicReferences.get(items[index].id)
+      );
+      noteIds.push(...this._db.notes.topicReferences.get(items[index].id));
```

Comment:
This can be greatly simplified. Right now it is a confusing mess.

1. use `for..of` loop
2. use variables to improve readability
3. No idea why `noteIds` is being used here?
4. Delete topic one-by-one instead of at the end. This will also take care of `lastNotebookId`.